http/logical.go

```diff
@@ -200,6 +200,61 @@ func handleLogicalInternal(core *vault.Core, injectDataIntoTopLevel bool) http.H
 			// requests for these paths always go to the token NS
 			case "auth/token/lookup-self", "auth/token/renew-self", "auth/token/revoke-self":
 				r = r.WithContext(newCtx)
+
+			// For the following operations, we can set the proper namespace context
+			// using the token's embedded nsID if a relative path was provided. Since
+			// this is done at the HTTP layer, the operation will still be gated by
+			// ACLs.
+			case "auth/token/renew", "auth/token/revoke", "auth/token/revoke-orphan":
+				token, ok := req.Data["token"]
```

Comment:
Note that if the path is auth/token/lookup, it will use the client token if "token" doesn't exist. So if token isn't there, you should probably use the request token instead, rather than bailing out.

Reply:
Added logic within the ok check to use the client token's ns and then break instead of breaking out immediately.